src/Features/Core/Portable/InitializeParameter/AbstractInitializeParameterCodeRefactoringProvider.cs

```diff
@@ -41,69 +47,102 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
             var position = context.Span.Start;
             var document = context.Document;
             var cancellationToken = context.CancellationToken;
-
             var syntaxTree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
             var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-
             var token = root.FindToken(position);
-            var parameterNode = GetParameterNode(token, position);
-            if (parameterNode == null)
-            {
-                return;
-            }
-
-            // Only offered when there isn't a selection, or the selection exactly selects
-            // a parameter name.
-            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
-            if (!context.Span.IsEmpty)
-            {
-                var parameterName = syntaxFacts.GetNameOfParameter(parameterNode);
-                if (parameterName == null || parameterName.Value.Span != context.Span)
-                {
-                    return;
-                }
-            }
-
-            var functionDeclaration = parameterNode.FirstAncestorOrSelf<SyntaxNode>(IsFunctionDeclaration);
-            if (functionDeclaration == null)
-            {
-                return;
-            }
-
-            var parameterDefault = syntaxFacts.GetDefaultOfParameter(parameterNode);
+            var firstparameterNode = GetParameterNode(token, position);
 
-            // Don't offer inside the "=initializer" of a parameter
-            if (parameterDefault?.Span.Contains(position) == true)
-            {
-                return;
-            }
+            // Creating a list and iterating it from the allows the right order or ifs
+            var parameterNodes = GetParameters(firstparameterNode);
+            var counter = 0;
 
-            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-
-            // we can't just call GetDeclaredSymbol on functionDeclaration because it could an anonymous function,
-            // so first we have to get the parameter symbol and then its containing method symbol
-            var parameter = (IParameterSymbol)semanticModel.GetDeclaredSymbol(parameterNode, cancellationToken);
-            if (parameter == null || parameter.Name == "")
-            {
-                return;
-            }
+            // List with parameterNodes that pass all checks
+            var listOfPotentiallyValidParametersNodes = new List<SyntaxNode>();
 
-            var method = (IMethodSymbol)parameter.ContainingSymbol;
-            if (method.IsAbstract ||
-                method.IsExtern ||
-                method.PartialImplementationPart != null ||
-                method.ContainingType.TypeKind == TypeKind.Interface)
+            if (!parameterNodes.IsDefault)
             {
-                return;
+                foreach (var parameterNode in parameterNodes)
```

Comment:
I think this loop needs to be rewritten. Right now we are doing things for each parameter that only needs to be done once. For example, the check for selection, checks on method symbol, and the call to `CanOfferRefactoring`.